Blockzilla/AddSearchEngineViewController.swift

```diff
@@ -142,25 +144,39 @@ class AddSearchEngineViewController: UIViewController, UITextViewDelegate {
         guard let name = nameInput.text else { return }
         guard let template = templateInput.text else { return }
         
+        if !AddSearchEngineViewController.isValidTemplate(template) || !searchEngineManager.isValidSearchEngineName(name) {
+            Toast(text: UIConstants.strings.errorTryAgain).show()
+            return
+        }
+        
         delegate.addSearchEngineViewController(self, name: name, searchTemplate: template)
         Toast(text: UIConstants.strings.NewSearchEngineAdded).show()
         self.navigationController?.popViewController(animated: true)
     }
     
     func textViewDidBeginEditing(_ textView: UITextView) {
-        templatePlaceholderLabel.isHidden = !isNullOrEmpty(text: textView.text)
+        templatePlaceholderLabel.isHidden = !AddSearchEngineViewController.isNullOrEmpty(text: textView.text)
     }
     
     func textViewDidChange(_ textView: UITextView) {
-        templatePlaceholderLabel.isHidden = !isNullOrEmpty(text: textView.text)
+        templatePlaceholderLabel.isHidden = !AddSearchEngineViewController.isNullOrEmpty(text: textView.text)
     }
     
     func textViewDidEndEditing(_ textView: UITextView) {
-        templatePlaceholderLabel.isHidden = !isNullOrEmpty(text: textView.text)
+        templatePlaceholderLabel.isHidden = !AddSearchEngineViewController.isNullOrEmpty(text: textView.text)
     }
     
-    private func isNullOrEmpty(text:String?) -> Bool{
+    private static func isNullOrEmpty(text:String?) -> Bool {
```

Comment:
Do we actually need this function? We're not actually using it with an optional string. 

Another shorthand for this can be `str?.isEmpty ?? true`. If we really want the method for readability/documentation we could do something like this:

```
extension Optional where Wrapped == String {
    var isNullOrEmpty: Bool { return self?.isEmpty ?? true }
}

str?.isNullOrEmpty
```